### PR TITLE
Feature ➤ Simple CameraX

### PR DESCRIPTION
**Pull request by EAlexisDuque**
## Description
Create a new class that allows you to use the new cameraX library with the design for the wallet application

## Trello Cards
* [Migrate to camera X [8D]](https://trello.com/c/H7RtwB7t/1281-migrate-to-camera-x-8d)

### CODE REVIEW

File: app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    private fun initCameraX() {
+        cameraFragment.startCamera()
+        captureButton?.setOnClickListener {
+            (captureButton?.drawable as? Animatable)?.start()
+            cameraFragment.takePhoto(cameraConfigObj.urlToSavePhoto)
+        }
+    }
+
+    override fun onCaptureSuccess(bitmap: Bitmap) {
```
Comment (cparracondor) [inline app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt]: I think it should be onImageCaptured

File: app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        showImageBitmap(bitmap)
+    }
+
+    override fun onImageSaved(bitmap: Bitmap) {
+        showImageBitmap(bitmap)
+    }
+
+    private fun showImageBitmap(bitmap: Bitmap) {
```
Comment (cparracondor) [inline app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt]: Just showImage no?

File: app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        val bitmapHeight = bitmap.height
+        val bitmapWidth = bitmap.width
+
+        if (bitmapWidth > bitmapHeight) {
```
Comment (cparracondor) [inline app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt]: I think you should create a function that explains this conditional.

File: app/src/main/res/layout/activity_camera.xml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+            <androidx.constraintlayout.widget.Barrier
+                android:id="@+id/descriptionAuxBarrier"
+                android:layout_width="@dimen/zero"
+                android:layout_height="wrap_content"
+                app:barrierDirection="bottom"
+                app:constraint_referenced_ids="capturePhotoDescription" />
+
+            <co.condorlabs.customcomponents.custombutton.CustomButton
+                android:id="@+id/cancelPhoto"
```
Comment (cparracondor) [inline app/src/main/res/layout/activity_camera.xml]: This views should start with bt -> btCancel

File: app/src/main/res/layout/activity_camera.xml [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                android:text="@string/retake_photo"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toStartOf="@id/cropPhoto"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                app:type="shapeWhite"
+                tools:visibility="visible" />
+
+            <androidx.appcompat.widget.AppCompatImageButton
+                android:id="@+id/captureButton"
```
Comment (cparracondor) [inline app/src/main/res/layout/activity_camera.xml]: same

File: app/src/main/res/layout/activity_camera.xml [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+                style="@style/camera_take_photo_button_style"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/descriptionAuxBarrier" />
+
+            <co.condorlabs.customcomponents.custombutton.CustomButton
+                android:id="@+id/cropPhoto"
```
Comment (cparracondor) [inline app/src/main/res/layout/activity_camera.xml]: same

File: app/src/main/res/values/colors.xml [hunk 7]
```diff
@@ -57,6 +57,7 @@
     <color name="emerald_disable_file_selector">#C9C9C9</color>
     <color name="emerald_enable_file_selector_icon">#5542A5F5</color>
 
-
-
+    <!-- WALLET CAMERA -->
+    <color name="background_black_with_with_40_percent_transparency">#66000000</color>
```
Comment (cparracondor) [inline app/src/main/res/values/colors.xml]: background_black_with_transparency

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -57,4 +57,13 @@
 
     <string name="hide">Hide</string>
     <string name="show">Show</string>
+
+    <!-- CAMERA -->
+    <string name="take_photo">take photo</string>
```
Comment (cparracondor) [inline app/src/main/res/values/strings.xml]: Probably something like

<string name="camera_photo_captured">Photo captured</string>
<string name="camera_retake_photo">Retake</string>

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/simplecamerax/CameraActivityTest.kt [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+            Assert.assertEquals(View.INVISIBLE, btnCancelVisibility)
+            Assert.assertEquals(View.INVISIBLE, btnCropCancelVisibility)
+            Assert.assertEquals(View.VISIBLE, btnCaptureVisibility)
+            Assert.assertEquals(View.VISIBLE, descriptionVisibility)
+        }
+    }
+
+    @Test
+    @LargeTest
+    fun shouldTakeAPhoto() {
```
Comment (cparracondor) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/simplecamerax/CameraActivityTest.kt]: Let's talk about this on Monday. Is the test really doing what it says?

File: app/build.gradle [hunk 10]
```diff
@@ -75,6 +82,9 @@ dependencies {
     implementation applicationDependencies.skeleton
 
     //Tests
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
[the remark this answers is not in the document]
Comment (EAlexisDuque) [reply app/build.gradle]: Yes

File: app/src/main/java/co/condorlabs/customcomponents/models/CameraConfig.kt [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+ * @company Condor Labs.
+ * @email eduque@condorlabs.io.
+ */
+@Parcelize
+data class CameraConfig(
+    val titleText: String? = null,
+    val descriptionText: String? = null,
+    val cancelButtonText: String? = null,
+    val cropButtonText: String? = null,
+    val urlToSavePhoto: String? = null
```
[the remark this answers is not in the document]
Comment (EAlexisDuque) [reply app/src/main/java/co/condorlabs/customcomponents/models/CameraConfig.kt]: Ok

File: app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+import co.condorlabs.customcomponents.R
+import co.condorlabs.customcomponents.models.CameraConfig
+import co.condorlabs.customcomponents.simplecamerax.fragment.SimpleCameraXFragment
+import kotlinx.android.synthetic.main.activity_camera.*
+import java.io.ByteArrayOutputStream
+
+class CameraActivity : AppCompatActivity(), SimpleCameraXFragment.OnCameraXListener {
+
+    private lateinit var cameraFragment: SimpleCameraXFragment
+    private lateinit var cameraConfigObj: CameraConfig
```
[the remark this answers is not in the document]
Comment (EAlexisDuque) [reply app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt]: Done

File: app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt (hunk 1 above)
Comment (EAlexisDuque) [reply app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt]: Ok

File: app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt (hunk 2 above)
Comment (EAlexisDuque) [reply app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt]: :)

File: app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt (hunk 3 above)
Comment (EAlexisDuque) [reply app/src/main/java/co/condorlabs/customcomponents/simplecamerax/CameraActivity.kt]: Ok

File: app/src/main/res/layout/activity_camera.xml (hunk 4 above)
Comment (EAlexisDuque) [reply app/src/main/res/layout/activity_camera.xml]: Ok

File: app/src/main/res/layout/activity_camera.xml (hunk 5 above)
Comment (EAlexisDuque) [reply app/src/main/res/layout/activity_camera.xml]: Done

File: app/src/main/res/layout/activity_camera.xml (hunk 6 above)
Comment (EAlexisDuque) [reply app/src/main/res/layout/activity_camera.xml]: Done

File: app/src/main/res/values/colors.xml (hunk 7 above)
Comment (EAlexisDuque) [reply app/src/main/res/values/colors.xml]: Done

File: app/src/main/res/values/strings.xml (hunk 8 above)
Comment (EAlexisDuque) [reply app/src/main/res/values/strings.xml]: Done

**Comment by feguerrero** [comment]
Version 80 already exists, change to 81

File: app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -57,4 +57,10 @@
 
     <string name="hide">Hide</string>
     <string name="show">Show</string>
+
+    <!-- CAMERA -->
+    <string name="crop_photo">Use photo</string>
+    <string name="retake_photo">Retake</string>
+    <string name="front_text">Front</string>
+    <string name="touch_to_take_photo">Touch to take photo</string>
```
[the remark this answers is not in the document]
Comment (EAlexisDuque) [reply app/src/main/res/values/strings.xml]: Done

File: app/build.gradle [hunk 14]
```diff
@@ -75,6 +82,8 @@ dependencies {
     implementation applicationDependencies.skeleton
 
     //Tests
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
```
[the remark this answers is not in the document]
Comment (EAlexisDuque) [reply app/build.gradle]: Done

**Comment by EAlexisDuque** [comment]
## Test
Screen Shot 2019-11-20 at 4 02 20 PM

**Pull request closed without merging**